Revert "Revert "merge: добавлен весь функционал""

ReTeRWert opened the pull request:
This reverts commit add3a0baa78993f61881e298d94b20aa3f1ce904.

ReTeRWert committed:
Revert "Revert "merge: добавлен весь функционал""

This reverts commit add3a0baa78993f61881e298d94b20aa3f1ce904.

krylov-sergey-sergeevich commented on src/main/java/ru/yandex/practicum/filmorate/dao/DirectorDbStorage.java:
```diff
@@ -0,0 +1,101 @@
+    public List<Director> findDirectorsByFilm(long filmId) {
+        String sql = "SELECT * " +
+                "FROM directors " +
+                "WHERE director_id = (SELECT director_id " +
```
с join тут никак нельзя запрос переделать?

krylov-sergey-sergeevich commented on src/main/java/ru/yandex/practicum/filmorate/dao/UserDbStorage.java:
```diff
@@ -84,48 +106,117 @@ public User findUserById(long id) {
     }
+
+    @Override
+    public void deleteUserById(Long userId) {
```
Вот тут точно так надо столько запросов? Через reference / cascad delete не удалятся ли другие сущности?

krylov-sergey-sergeevich reviewed (changes requested):
Привет

**Логи**

1) Логи все должны быть на англ языке

2) Вот вы выбрали алгоритмы которые хотите реализовать, я как ревьювер могу искать и проверять что все есть, но в реальном мире все алгоритмы имеют название/номер чаще и то и то, например у меня на работе дают названия на подобии TS-1 TS-2 TS-3 в документации соответственно есть эти же номера, а далее в коде в логах мы пишем "TS-1 создание того того" и если у кого-то есть вопрос по реализации он идет в документацию и ищет по номеру алгоритма. Итого: Просьба в реадми + в описание PR пронумеровать алгоритмы и добавить в логах соответствующие номера.

3) Так -же в readme следует указать кто какие алгоритмы реализовывал.

4) Во всех методах контроллеров обязательно должны быть логи

Логика кодовая

1) Если где-то есть подзапросы в запросах , просьба еще раз глянуть и подумать нельзя ли там использовать join

Буду ждать с правками и ответами на вопросы в коде =)

84R5 committed:
<ref> delete InMemoryFilmStorage.java
<ref> findDirectorsByFilm > change (SELECT to JOIN)

84R5 committed:
<add> CASCADE DELETE

84R5 committed:
<fix> change Integer to Long

84R5 committed:
<fix> change Integer to Long
<ref> delete Mapper, move to Storage

84R5 committed:
<fix> fix test add .friends(new HashSet<>()) [] to expected:

mvinogradovHub committed:
1. Перевести логи на английский
2. Навесить логи на все методы во всех контроллерах
3. Навесить логи в обработку исключений сервисов

Commit:
fix: поменял получение популярных фильмов

vvalferov commented:
1. [FT-1] Функциональность "Лента событий" - Михаил (mvinogradovHub)
2. [FT-2] Функциональность "Поиск" - Игорь (MR-k0f31n)
3. [FT-3] Функциональность "Вывод самых популярных фильмов по жанру и годам" - Руслан (NiceTulipe)
4. [FT-4] Функциональность "Удаление фильмов и пользователей" - Денис (ReTerWert)
5. [FT-5] Функциональность "Добавление режиссёров в фильмы" - Денис (ReTerWert)
6. [FT-6] Функциональность "Общие фильмы" - Денис (ReTerWert)
7. [FT-7] Функциональность "Рекомендации" - Александр (84R5)
8. [FT-8] Функциональность "Отзывы" - Денис (ReTerWert)

krylov-sergey-sergeevich commented on src/main/java/ru/yandex/practicum/filmorate/controllers/UserController.java:
```diff
@@ -26,57 +28,73 @@ public UserController(UserService userService) {
 
+    @GetMapping("/{id}/feed")
+    public Collection<Feed> getFeed(@PathVariable Long id)
+            throws RuntimeException {
```
throws RuntimeException - зачем писать если можно без этого?

krylov-sergey-sergeevich commented on src/main/java/ru/yandex/practicum/filmorate/controllers/UserController.java:
```diff
@@ -26,57 +28,73 @@ public UserController(UserService userService) {
     }
 
+    @GetMapping("/{id}/feed")
+    public Collection<Feed> getFeed(@PathVariable Long id)
```
Лучше List

krylov-sergey-sergeevich commented on src/main/java/ru/yandex/practicum/filmorate/exception/ErrorHandler.java:
```diff
@@ -0,0 +1,31 @@
+import java.util.Map;
+
+@RestControllerAdvice
+public class ErrorHandler {
```
Лучше на англ писать текста

krylov-sergey-sergeevich commented on src/main/java/ru/yandex/practicum/filmorate/service/review/ReviewService.java:
```diff
@@ -0,0 +1,142 @@
+    public Review createReview(Review review) {
+        if (review.getReviewId() != null) {
+            log.warn("Review already exists");
+            throw new ValidationException("Ревью уже существует");
```
russian

krylov-sergey-sergeevich commented on src/main/java/ru/yandex/practicum/filmorate/service/review/ReviewService.java:
```diff
@@ -0,0 +1,142 @@
+            throw new ValidationException("Review not found");
+        } else {
+            Review reviewInStorage = reviewDbStorage.updateReview(review);
+            feedStorage.addFeed(Feed.builder().operation(FeedOperation.UPDATE).eventType(FeedEventType.REVIEW).entityId(reviewInStorage.getReviewId()).userId(reviewInStorage.getUserId()).build());
```
В одну строку такое выглядит страшновато

krylov-sergey-sergeevich reviewed (approved):
Привет, вижу правки - зачет!
